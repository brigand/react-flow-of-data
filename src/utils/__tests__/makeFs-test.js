const nodeFs = require('fs');
const path = require('path');
const makeFs = require('../makeFs.js');

const fixtureFs = path.resolve(__dirname, '..', '__fixtures__');

it(`works`, () => {
  makeFs(`/tmp/`);
});

it(`reads a file`, () => {
  return makeFs(fixtureFs).readP('foo/x.txt').then((c) => {
    expect(c).toBe(`x`);
  });
});

it(`reads a directory`, () => {
  return makeFs(fixtureFs).readDirP('.', 'txt').then((files) => {
    expect(files).toMatchSnapshot();
  });
});

