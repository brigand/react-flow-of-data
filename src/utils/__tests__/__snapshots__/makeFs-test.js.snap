// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reads a directory 1`] = `
Array [
  Object {
    "content": "",
    "loc": "/Users/fb/github/brigand/react-flow-of-data/src/utils/__fixtures__/foo/bar/y.txt",
    "meta": undefined,
    "name": "y.txt",
  },
  Object {
    "content": "xâ€¦",
    "loc": "/Users/fb/github/brigand/react-flow-of-data/src/utils/__fixtures__/foo/x.txt",
    "meta": undefined,
    "name": "x.txt",
  },
]
`;
